### ISSUE TITLE
Proper form input for deferred execution

### ISSUE THREAD

**Issue by lukasjuhrich** [agdsn/pycroft#396]
Currently, there are two separate inputs for actions like moving a user out: a checkbox for immediate execution and an optional date field. However, if one enters a date and _still_ checks the checkbox, the date is not respected.

This should be solved by a UI mechanic not allowing for such inconsistent submissions.

**Comment by MarauderXtreme** [comment]
I beg to differ.
This behaviour needs to be addressed in backend as well.
Frontend-validation is too easily exploitable.

**Comment by lukasjuhrich** [comment]
Yes and no: In the specific example given above, we're not talking about a validation mechanism protecting _the backend_ from bad data, but protecting _the user_ from submitting things he most likely did not intend. By that logic, a bike helmet would be „too easily exploitable“ because a person could just choose not to wear it.

On the other hand, there is a different scenario where that would be wise: entering a past date. Jugding by the code, this should already yield a ValueError which is essentially backend validation (see [here](https://github.com/agdsn/pycroft/blob/9b8a0eb86dfd00750f554161f3d37764cd338a58/pycroft/lib/task.py#L168)), but as I just tested with myself, this is not respected: I got moved out immediately.

So yes, ensuring that the backend validation works for past dates should be adressed as well, and in that breath we can validate the exclusivity of immedate vs. date given as well.

**Comment by ibot3** [comment]
Entering a past date is actually a quite useful feature and should not be prohibited.
I already did some work on a branch which allows it to hide form fields if a checkbox gets checked.

**Comment by lukasjuhrich** [comment]
I actually find that behavior very, very confusing – this is something the UI should communicate. I thought this was undefined behavior and expected an error, and I guess a significant portion (perhaps not the majority, but that's not the point) of first-time users will assume the same.

I've thought about this for some time, and come up with the following solution:
- Hide the checkbox and introduce a help-text area  concisely describing the current state (like „Execution: *Retroactive*. Connection terminated immediately, end of membership backdated“)
- Hook the [changedate event](https://bootstrap-datepicker.readthedocs.io/en/latest/events.html#changedate) of the datepicker to change the helptext – and check and deactivate the “end membership as well” hook when in the retroactive mode, because backdating doesn't make sense if there isn't a membership end to backdate.
- Submit the data just as we did before, only with the hidden „immediate“ checkbox being hidden and auto-set by the changedate event handler.

For the help text, I can imagine something deemphasized along the lines of the remark you see under this very input field:
![image](https://user-images.githubusercontent.com/10060406/88528000-513fcd00-cffe-11ea-932b-3167bbc6d5da.png)
The „info“ icon would be a perfect suit.
